PyOpenGL 3.1.3b2 on main breaks vispy

I have vispy travis CI jobs running weekly (with cron) and they have started failing with the newest conda-forge release of pyopengl 3.1.3b2. I created another branch that forces the version of pyopengl to 3.1.1a1 and all tests pass. The failing tests have the following error:

```
self = <OpenGL.converters.SizedOutputOrInput object at 0x2b7757b86af0>
result = None
baseOperation = <OpenGL.platform.baseplatform.glGetBufferParameteriv object at 0x2b7757b0c7e8>
pyArgs = (GL_ARRAY_BUFFER, GL_BUFFER_SIZE, <object object at 0x2b7755acf6e0>)
cArgs = (GL_ARRAY_BUFFER, GL_BUFFER_SIZE, array([192], dtype=int32))
    def oldStyleReturn( self, result, baseOperation, pyArgs, cArgs ):
        """Retrieve cArgs[ self.index ]"""
>       result = cArgs[ self.outIndex ]
E       AttributeError: ('outIndex', <bound method SizedOutputOrInput.oldStyleReturn of <OpenGL.converters.SizedOutputOrInput object at 0x2b7757b86af0>>, (None, <OpenGL.platform.baseplatform.glGetBufferParameteriv object at 0x2b7757b0c7e8>, (GL_ARRAY_BUFFER, GL_BUFFER_SIZE, <object object at 0x2b7755acf6e0>), (GL_ARRAY_BUFFER, GL_BUFFER_SIZE, array([192], dtype=int32))), {})
../../../anaconda/envs/testenv/lib/python2.7/site-packages/OpenGL/converters.py:202: AttributeError
```

You can see the failing jobs [here](https://travis-ci.org/vispy/vispy/builds/482631153) and the new succeeding jobs with 3.1.1a1 [here](https://travis-ci.org/vispy/vispy).

Any ideas? Where are these new versions coming from? I can't find a web page that has these new downloads or a source distribution on sourceforge or...anywhere.

@djhoese The error in the first breaking (pyopengl) commit is within [this function](https://github.com/vispy/vispy/blob/master/vispy/gloo/gl/_pyopengl2.py#L89). Obviously vispy is adding to much parameters or the like. Hope you can get something out of it.

```bash
――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――― test_functionality_pyopengl ―――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――

args = (), kwargs = {}

    def skipper_func(*args, **kwargs):
        """Skipper for normal test functions."""
        if skip_val():
            raise SkipTest(get_msg(f, msg))
        else:
>           return f(*args, **kwargs)

../../../miniconda3/envs/pyopengl/lib/python3.6/site-packages/numpy/testing/_private/decorators.py:155: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
../../../miniconda3/envs/pyopengl/lib/python3.6/site-packages/numpy/testing/_private/decorators.py:155: in skipper_func
    return f(*args, **kwargs)
../vispy-stuff/vispy/vispy/gloo/gl/tests/test_functionality.py:60: in test_functionality_pyopengl
    _test_functionality('pyopengl2')
../vispy-stuff/vispy/vispy/gloo/gl/tests/test_functionality.py:94: in _test_functionality
    objects = _prepare_vis()
../vispy-stuff/vispy/vispy/gloo/gl/tests/test_functionality.py:289: in _prepare_vis
    name, count, type = gl.glGetActiveAttrib(hprog, i)
../vispy-stuff/vispy/vispy/gloo/gl/_pyopengl2.py:96: in glGetActiveAttrib
    GL.glGetActiveAttrib(program, index, bufsize, length, size, type, name)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = OpenGL.lazywrapper.lazy( 'glGetActiveAttrib' )
args = (1, 0, 256, <OpenGL.converters.c_int_Array_1 object at 0x7f53dd617730>, <OpenGL.converters.c_int_Array_1 object at 0x7f53dd6178c8>, <vispy.gloo.gl._pyopengl2.c_uint_Array_1 object at 0x7f53dd617950>, ...)
named = {}

    def __call__( self, *args, **named ):
        """returns self.wrapperFunction( self.baseFunction, *args, **named )"""
>       return self.wrapperFunction( self.baseFunction, *args, **named )
E       TypeError: glGetActiveAttrib() takes from 3 to 4 positional arguments but 8 were given

../../../miniconda3/envs/pyopengl/lib/python3.6/site-packages/OpenGL/latebind.py:61: TypeError

```

--- a/.azure-pipelines/azure-pipelines-linux.yml
+++ b/.azure-pipelines/azure-pipelines-linux.yml
@@ -1,5 +1,9 @@
+# This file was generated automatically from conda-smithy. To update this configuration,
+# update the conda-forge.yml and/or the recipe/meta.yaml.
+# -*- mode: yaml -*-
+
 jobs:
-- job: linux_64
+- job: linux
   pool:
     vmImage: ubuntu-16.04
   timeoutInMinutes: 240
@@ -8,11 +12,22 @@ jobs:
     matrix:
       linux_:
         CONFIG: linux_
+        UPLOAD_PACKAGES: False
   steps:
   - script: |
       sudo pip install --upgrade pip
       sudo pip install setuptools shyaml
     displayName: Install dependencies
 
+  # configure qemu binfmt-misc running.  This allows us to run docker containers 
+  # embedded qemu-static
+  - script: |
+      docker run --rm --privileged multiarch/qemu-user-static:register
+      ls /proc/sys/fs/binfmt_misc/
+    condition: not(startsWith(variables['CONFIG'], 'linux_64'))
+    displayName: Configure binfmt_misc
+
   - script: .azure-pipelines/run_docker_build.sh
     displayName: Run docker build
+    env:
+      BINSTAR_TOKEN: $(BINSTAR_TOKEN)

--- a/.azure-pipelines/build_steps.sh
+++ b/.azure-pipelines/build_steps.sh
@@ -24,15 +24,15 @@ conda install --yes --quiet conda-forge-ci-setup=2 conda-build -c conda-forge
 # set up the condarc
 setup_conda_rc "${FEEDSTOCK_ROOT}" "${RECIPE_ROOT}" "${CONFIG_FILE}"
 
-# A lock sometimes occurs with incomplete builds. The lock file is stored in build_artifacts.
-conda clean --lock
-
-run_conda_forge_build_setup# make the build number clobber
+run_conda_forge_build_setup
+# make the build number clobber
 make_build_number "${FEEDSTOCK_ROOT}" "${RECIPE_ROOT}" "${CONFIG_FILE}"
 
 conda build "${RECIPE_ROOT}" -m "${CI_SUPPORT}/${CONFIG}.yaml" \
     --clobber-file "${CI_SUPPORT}/clobber_${CONFIG}.yaml"
 
-
+if [[ "${UPLOAD_PACKAGES}" != "False" ]]; then
+    upload_package "${FEEDSTOCK_ROOT}" "${RECIPE_ROOT}" "${CONFIG_FILE}"
+fi
 
 touch "/home/conda/feedstock_root/build_artifacts/conda-forge-build-done-${CONFIG}"

--- a/.azure-pipelines/run_docker_build.sh
+++ b/.azure-pipelines/run_docker_build.sh
@@ -33,7 +33,7 @@ if [ -z "$CONFIG" ]; then
 fi
 
 pip install shyaml
-DOCKER_IMAGE=$(cat "${FEEDSTOCK_ROOT}/.ci_support/${CONFIG}.yaml" | shyaml get-value docker_image.0 condaforge/linux-anvil )
+DOCKER_IMAGE=$(cat "${FEEDSTOCK_ROOT}/.ci_support/${CONFIG}.yaml" | shyaml get-value docker_image.0 condaforge/linux-anvil-comp7 )
 
 mkdir -p "$ARTIFACTS"
 DONE_CANARY="$ARTIFACTS/conda-forge-build-done-${CONFIG}"
@@ -41,13 +41,14 @@ rm -f "$DONE_CANARY"
 # Not all providers run with a real tty.  Disable using one
 DOCKER_RUN_ARGS=" "
 
-
+export UPLOAD_PACKAGES="${UPLOAD_PACKAGES:-True}"
 docker run ${DOCKER_RUN_ARGS} \
            -v "${RECIPE_ROOT}":/home/conda/recipe_root:ro,z \
            -v "${FEEDSTOCK_ROOT}":/home/conda/feedstock_root:rw,z \
            -e CONFIG \
            -e BINSTAR_TOKEN \
            -e HOST_USER_ID \
+           -e UPLOAD_PACKAGES \
            $DOCKER_IMAGE \
            bash \
            /home/conda/feedstock_root/${PROVIDER_DIR}/build_steps.sh

--- a/.ci_support/linux_.yaml
+++ b/.ci_support/linux_.yaml
@@ -5,7 +5,7 @@ channel_sources:
 channel_targets:
 - conda-forge main
 docker_image:
-- condaforge/linux-anvil
+- condaforge/linux-anvil-comp7
 pin_run_as_build:
   python:
     min_pin: x.x

--- a/.circleci/build_steps.sh
+++ b/.circleci/build_steps.sh
@@ -24,9 +24,6 @@ conda install --yes --quiet conda-forge-ci-setup=2 conda-build -c conda-forge
 # set up the condarc
 setup_conda_rc "${FEEDSTOCK_ROOT}" "${RECIPE_ROOT}" "${CONFIG_FILE}"
 
-# A lock sometimes occurs with incomplete builds. The lock file is stored in build_artifacts.
-conda clean --lock
-
 source run_conda_forge_build_setup
 
 # make the build number clobber
@@ -35,8 +32,8 @@ make_build_number "${FEEDSTOCK_ROOT}" "${RECIPE_ROOT}" "${CONFIG_FILE}"
 conda build "${RECIPE_ROOT}" -m "${CI_SUPPORT}/${CONFIG}.yaml" \
     --clobber-file "${CI_SUPPORT}/clobber_${CONFIG}.yaml"
 
-
-upload_package "${FEEDSTOCK_ROOT}" "${RECIPE_ROOT}" "${CONFIG_FILE}"
-
+if [[ "${UPLOAD_PACKAGES}" != "False" ]]; then
+    upload_package "${FEEDSTOCK_ROOT}" "${RECIPE_ROOT}" "${CONFIG_FILE}"
+fi
 
 touch "/home/conda/feedstock_root/build_artifacts/conda-forge-build-done-${CONFIG}"

--- a/.circleci/build_steps.sh
+++ b/.circleci/build_steps.sh
@@ -26,6 +26,14 @@ setup_conda_rc "${FEEDSTOCK_ROOT}" "${RECIPE_ROOT}" "${CONFIG_FILE}"
 
 source run_conda_forge_build_setup
 
+
+# Install the yum requirements defined canonically in the
+# "recipe/yum_requirements.txt" file. After updating that file,
+# run "conda smithy rerender" and this line will be updated
+# automatically.
+/usr/bin/sudo -n yum install -y mesa-libGL
+
+
 # make the build number clobber
 make_build_number "${FEEDSTOCK_ROOT}" "${RECIPE_ROOT}" "${CONFIG_FILE}"
 

--- a/.circleci/config.yml
+++ b/.circleci/config.yml
@@ -1,3 +1,7 @@
+# This file was generated automatically from conda-smithy. To update this configuration,
+# update the conda-forge.yml and/or the recipe/meta.yaml.
+# -*- mode: yaml -*-
+
 version: 2
 
 jobs:
@@ -14,7 +18,7 @@ jobs:
             ./.circleci/fast_finish_ci_pr_build.sh
             ./.circleci/checkout_merge_commit.sh
       - run:
-          command: docker pull condaforge/linux-anvil
+          command: docker pull condaforge/linux-anvil-comp7
       - run:
           # Run, test and (if we have a BINSTAR_TOKEN) upload the distributions.
           command: ./.circleci/run_docker_build.sh

--- a/.circleci/fast_finish_ci_pr_build.sh
+++ b/.circleci/fast_finish_ci_pr_build.sh
@@ -1,4 +1,4 @@
 #!/bin/bash
 
-curl https://raw.githubusercontent.com/conda-forge/conda-forge-ci-setup-feedstock/branch2.0/recipe/conda_forge_ci_setup/ff_ci_pr_build.py | \
+curl https://raw.githubusercontent.com/conda-forge/conda-forge-ci-setup-feedstock/master/recipe/conda_forge_ci_setup/ff_ci_pr_build.py | \
      python - -v --ci "circle" "${CIRCLE_PROJECT_USERNAME}/${CIRCLE_PROJECT_REPONAME}" "${CIRCLE_BUILD_NUM}" "${CIRCLE_PR_NUMBER}"

--- a/.circleci/run_docker_build.sh
+++ b/.circleci/run_docker_build.sh
@@ -33,7 +33,7 @@ if [ -z "$CONFIG" ]; then
 fi
 
 pip install shyaml
-DOCKER_IMAGE=$(cat "${FEEDSTOCK_ROOT}/.ci_support/${CONFIG}.yaml" | shyaml get-value docker_image.0 condaforge/linux-anvil )
+DOCKER_IMAGE=$(cat "${FEEDSTOCK_ROOT}/.ci_support/${CONFIG}.yaml" | shyaml get-value docker_image.0 condaforge/linux-anvil-comp7 )
 
 mkdir -p "$ARTIFACTS"
 DONE_CANARY="$ARTIFACTS/conda-forge-build-done-${CONFIG}"
@@ -41,13 +41,14 @@ rm -f "$DONE_CANARY"
 # Enable running in interactive mode attached to a tty
 DOCKER_RUN_ARGS=" -it "
 
-
+export UPLOAD_PACKAGES="${UPLOAD_PACKAGES:-True}"
 docker run ${DOCKER_RUN_ARGS} \
            -v "${RECIPE_ROOT}":/home/conda/recipe_root:ro,z \
            -v "${FEEDSTOCK_ROOT}":/home/conda/feedstock_root:rw,z \
            -e CONFIG \
            -e BINSTAR_TOKEN \
            -e HOST_USER_ID \
+           -e UPLOAD_PACKAGES \
            $DOCKER_IMAGE \
            bash \
            /home/conda/feedstock_root/${PROVIDER_DIR}/build_steps.sh

--- a/README.md
+++ b/README.md
@@ -1,7 +1,9 @@
+<!--
+# -*- mode: jinja -*-
+-->
+
 About pyopengl
 ==============
-
-[![Powered by NumFOCUS](https://img.shields.io/badge/powered%20by-NumFOCUS-orange.svg?style=flat&colorA=E1523D&colorB=007D8A)](http://numfocus.org)
 
 Home: http://pyopengl.sourceforge.net
 
@@ -50,6 +52,8 @@ conda search pyopengl --channel conda-forge
 
 About conda-forge
 =================
+
+[![Powered by NumFOCUS](https://img.shields.io/badge/powered%20by-NumFOCUS-orange.svg?style=flat&colorA=E1523D&colorB=007D8A)](http://numfocus.org)
 
 conda-forge is a community-led conda channel of installable packages.
 In order to provide high-quality builds, the process has been automated into the

--- a/azure-pipelines.yml
+++ b/azure-pipelines.yml
@@ -1,4 +1,6 @@
+# This file was generated automatically from conda-smithy. To update this configuration,
+# update the conda-forge.yml and/or the recipe/meta.yaml.
+# -*- mode: yaml -*-
+
 jobs:
   - template: ./.azure-pipelines/azure-pipelines-linux.yml
-
-  

--- a/recipe/meta.yaml
+++ b/recipe/meta.yaml
@@ -8,11 +8,13 @@ source:
   fn: PyOpenGL-{{ version }}.tar.gz
   url: https://pypi.python.org/packages/source/P/PyOpenGL/PyOpenGL-{{ version }}.tar.gz
   sha256: ac73b4c0b433be02d6a55b9ed768acf07003a92a3315ba9d62bc8f8da5f0c291
+  patches:
+    - remove_redefinition_of_glGetBufferParameteriv-wrapper.patch
 
 build:
   noarch: python
   script: python -m pip install --no-deps --ignore-installed .
-  number: 0
+  number: 1
 
 requirements:
   host:

--- a/recipe/remove_redefinition_of_glGetBufferParameteriv-wrapper.patch
+++ b/recipe/remove_redefinition_of_glGetBufferParameteriv-wrapper.patch
@@ -1,0 +1,17 @@
+Index: OpenGL/GL/VERSION/GL_1_5.py
+IDEA additional info:
+Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
+<+>UTF-8
+===================================================================
+--- OpenGL/GL/VERSION/GL_1_5.py	(revision bf11e7e59bff4a57dffbe79e1b27dac5a09166fc)
++++ OpenGL/GL/VERSION/GL_1_5.py	(revision abb9d9f2c7fae4d5b700e5f70e2ada62eb9211f9)
+@@ -124,9 +124,6 @@
+         size = ArrayDatatype.arrayByteCount( data )
+     return baseOperation( target, offset, size, data )
+ 
+-glGetBufferParameteriv = wrapper.wrapper(glGetBufferParameteriv).setOutput(
+-    "params",(1,), orPassIn=True
+-)
+ @_lazy( glGetBufferPointerv )
+ def glGetBufferPointerv( baseOperation, target, pname, params=None ):
+     """Retrieve a ctypes pointer to buffer's data"""

--- a/recipe/yum_requirements.txt
+++ b/recipe/yum_requirements.txt
@@ -1,0 +1,1 @@
+mesa-libGL

